File: core/src/ru/alexkulikov/firstfame/background/Mountain.java
package ru.alexkulikov.firstfame.background;

import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.scenes.scene2d.Actor;

import ru.alexkulikov.firstfame.FadeActor;
import ru.alexkulikov.firstfame.TextureLoader;

import static ru.alexkulikov.firstfame.objects.Constants.*;

public class Mountain extends FadeActor {
    private Sprite sprite;
    float scalableY;

    public Mountain() {
        setBounds(0, 0, 0, 0);
        sprite = new Sprite(TextureLoader.getMountain());
        scalableY = VIEWPORT_WIDTH / RATIO_STANDARD;
        sprite.setBounds(0, 0, 0, 0);
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        super.draw(batch, parentAlpha);
        batch.draw(sprite, sprite.getX(), 0.0f, scalableY, VIEWPORT_WIDTH);
    }

    public void setSpriteX(float x) {
        sprite.setBounds(x, 0, scalableY, VIEWPORT_WIDTH);
    }

    public float getSpriteOffset() {
        return sprite.getX() + sprite.getWidth();
    }

    public float getSpriteWidth() {
        return scalableY;
    }
}

File: core/src/ru/alexkulikov/firstfame/background/Mountain.kt
package ru.alexkulikov.firstfame.background

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.Sprite
import com.badlogic.gdx.scenes.scene2d.Actor

import ru.alexkulikov.firstfame.FadeActor
import ru.alexkulikov.firstfame.TextureLoader

import ru.alexkulikov.firstfame.objects.Constants.*

class Mountain : FadeActor() {
    private val spriteWidth = VIEWPORT_WIDTH / RATIO_STANDARD
    val sprite: Sprite = Sprite(TextureLoader.getMountain())

    val spriteOffset: Float
        get() = sprite.x + sprite.width

    init {
        sprite.setBounds(0f, 0f, spriteWidth, VIEWPORT_WIDTH.toFloat())
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        super.draw(batch, parentAlpha)
        batch.draw(sprite, sprite.x, 0.0f, spriteWidth, VIEWPORT_WIDTH.toFloat())
    }

    fun setSpriteX(x: Float) {
        sprite.setBounds(x, 0f, spriteWidth, VIEWPORT_WIDTH.toFloat())
    }
}
